### PR TITLE
[SmartTrak import] Fix tank count crash

### PR DESCRIPTION
**Pull request by salvadorcunat**
### Describe the pull request:

- [X] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

After changes introduced in commits 7c9f46a and 794066b, the import crashes due to get_cylinder() resulting in a NULL pointer as we try to parse a number of tanks greater than the one reported by libdc (the previously allocated number of 10 is overriden while doing libdc parsing).
Using get_or_create_cylinder() instead (and following a check of the cylinder) solves the issue.

### Changes made:

1) Solve reported crash.
2) Add other unrelated sanity checks (just in case).

### CODE REVIEW

File: smtk-import/smartrak.c [hunk 1]
```diff
@@ -44,7 +44,8 @@ static void smtk_free(char **array, int count)
 {
 	int n;
 	for (n = 0; n < count; n++)
-		free(array[n]);
+		if (array[n])
```
Comment (bstoeger) [inline smtk-import/smartrak.c]: This seems unnecessary - free() already tests for NULL.

**Review by bstoeger — COMMENTED** [review]
Looks good to me - thanks.

Only one minor comment and a coding-style note:
Usually we don't put a space after the logical negation operator `!`.

File: smtk-import/smartrak.c [hunk 2]
```diff
@@ -1038,24 +1036,27 @@ void smartrak_import(const char *file, struct dive_table *divetable)
 		int tankidxcol = coln(TANKIDX);
 
 		for (i = 0; i < tanks; i++) {
-			if (get_cylinder(smtkdive, i)->start.mbar == 0)
-				get_cylinder(smtkdive, i)->start.mbar = lrint(strtod(col[(i * 2) + pstartcol]->bind_ptr, NULL) * 1000);
+			cylinder_t *tmptank = get_or_create_cylinder(smtkdive, i);
+			if (! tmptank)
```
Comment (dirkhh) [inline smtk-import/smartrak.c]: as mentioned, we usually would write 
`if (!tmptank)`

File: smtk-import/smartrak.c [hunk 3]
```diff
@@ -1038,24 +1036,27 @@ void smartrak_import(const char *file, struct dive_table *divetable)
...
 			 * If there is a start pressure ensure that end pressure is not zero as
 			 * will be registered in DCs which only keep track of differential pressures,
 			 * and collect the data registered  by the user in mdb
 			 */
-			if (get_cylinder(smtkdive, i)->end.mbar == 0 && get_cylinder(smtkdive, i)->start.mbar != 0)
-					get_cylinder(smtkdive, i)->end.mbar = lrint(strtod(col[(i * 2) + 1 + pstartcol]->bind_ptr, NULL) * 1000 ? : 1000);
-			if (get_cylinder(smtkdive, i)->gasmix.o2.permille == 0)
-				get_cylinder(smtkdive, i)->gasmix.o2.permille = lrint(strtod(col[i + o2fraccol]->bind_ptr, NULL) * 10);
+			if (tmptank->end.mbar == 0 && tmptank->start.mbar != 0)
+					tmptank->end.mbar = lrint(strtod(col[(i * 2) + 1 + pstartcol]->bind_ptr, NULL) * 1000 ? : 1000);
```
Comment (dirkhh) [inline smtk-import/smartrak.c]: whitespace? This seems to have one too many indentations

**Review by dirkhh — COMMENTED** [review]
Salvador, I think it would be nice to clean up the small number of nit-picks from Berthold and me...

**Comment by salvadorcunat** [comment]
Have just sent a commit to fix them.
Thanks Berthold and Dirk.

**Review by dirkhh — APPROVED** [review]
thanks - this works

**Pull request merged**